test/solidity_dynamic_strings.js

```diff
@@ -0,0 +1,54 @@
+const Web3 = require("web3");
+const assert = require("assert");
+const Ganache = require("../index");
+const path = require("path");
+const compileAndDeploy = require("./helpers/contracts").compileAndDeploy;
+
+const mnemonic = "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat";
+
+function setUp(options = { mnemonic }, contractName = "DynamicStringLength") {
+  const context = {
+    options: options,
+    provider: null,
+    web3: null,
+    accounts: [],
+    contractArtifact: {},
+    instance: null
+  };
+
+  before("setup web3", async function() {
+    context.provider = Ganache.provider(context.options);
+    context.options.blockTime = 2000;
+    context.web3 = new Web3(context.provider);
+  });
+
+  before("compile source", async function() {
+    this.timeout(10000);
+    context.contractArtifact = await compileAndDeploy(
+      path.join(__dirname, ".", `${contractName}.sol`),
+      contractName,
+      context.web3
+    );
+    context.instance = context.contractArtifact.instance;
+  });
+  return context;
+}
+
+describe("Solidity variable string length", function() {
+  const context = setUp();
```

Comment:
readability: since this is the only place the `setUp` function is called, I'm not sure what value it's providing. The usual flow for mocha tests is as shown in the snippet below, and this breaks that flow

```
describe(...
  describe( ...
    before(...
    it(...
    after(...
  }
}
```

Reply:
Great suggestions for overall refactoring efforts.